cmd/lotus-shed/provider.go

```diff
@@ -0,0 +1,183 @@
+					Usage:   "The layer name for this data push. 'base' is recommended for single-miner setup. Overwrites.",
+				},
+			},
+			Action: fromMiner,
+		},
+	},
+}
+
+const (
+	FlagMinerRepo = "miner-repo"
+)
+
+const FlagMinerRepoDeprecation = "storagerepo"
+
+func fromMiner(cctx *cli.Context) error {
+	ctx := context.Background()
+
+	r, err := repo.NewFS(cctx.String(FlagMinerRepo))
+	if err != nil {
+		return err
+	}
+
+	ok, err := r.Exists()
+	if err != nil {
+		return err
+	}
+
+	if !ok {
+		return fmt.Errorf("repo not initialized")
+	}
+
+	lr, err := r.LockRO(repo.StorageMiner)
+	if err != nil {
+		return fmt.Errorf("locking repo: %w", err)
+	}
+	defer func() { _ = lr.Close() }()
+
+	cfgNode, err := lr.Config()
+	if err != nil {
+		return fmt.Errorf("getting node config: %w", err)
+	}
+	smCfg := cfgNode.(*config.StorageMiner)
+
+	db, err := harmonydb.NewFromConfig(smCfg.HarmonyDB)
+	if err != nil {
+		return fmt.Errorf("could not reach the database. Ensure the Miner config toml's HarmonyDB entry"+
+			" is setup to reach Yugabyte correctly: %w", err)
+	}
+
+	var titles []string
+	err = db.Select(ctx, &titles, `SELECT title FROM harmony_config`)
+	if err != nil {
+		return fmt.Errorf("miner cannot reach the db. Ensure the config toml's HarmonyDB entry"+
+			" is setup to reach Yugabyte correctly: %s", err.Error())
+	}
+	name := cctx.String("layer")
+	if name == "" {
+		name = fmt.Sprintf("mig%d", len(titles))
+	}
+	msg := "Layer " + name + ` created. `
+
+	// Copy over identical settings:
+
+	buf, err := os.ReadFile(path.Join(lr.Path(), "config.toml"))
+	if err != nil {
+		return fmt.Errorf("could not read config.toml: %w", err)
+	}
+	var lpCfg config.LotusProviderConfig
+	_, err = toml.Decode(string(buf), &lpCfg)
+	if err != nil {
+		return fmt.Errorf("could not decode toml: %w", err)
+	}
+
+	// Populate Miner Address
+	sm, cc, err := cliutil.GetStorageMinerAPI(cctx)
+	if err != nil {
+		return fmt.Errorf("could not get storageMiner API: %w", err)
+	}
+	defer cc()
+	addr, err := sm.ActorAddress(ctx)
+	if err != nil {
+		return fmt.Errorf("could not read actor address: %w", err)
+	}
+
+	lpCfg.Addresses.MinerAddresses = []string{addr.String()}
+
+	// TODO Apis.StorageSecret  storage repo and reading the rpc secret
+	//lr.??
+
+	// Populate API Key
+	_, header, err := cliutil.GetRawAPI(cctx, repo.FullNode, "v0")
+	if err != nil {
+		return fmt.Errorf("cannot read API: %w", err)
+	}
+
+	lpCfg.Apis.FULLNODE_API_INFO = []string{header.Get("Authorization")[7:]}
+
+	// Enable WindowPoSt
+	lpCfg.Subsystems.EnableWindowPost = true
+	msg += `\nBefore running lotus-provider, ensure any miner/worker answering of WindowPost is disabled.\n`
+
+	// Express as configTOML
+	configTOML := &bytes.Buffer{}
+	if err = toml.NewEncoder(configTOML).Encode(lpCfg); err != nil {
+		return err
+	}
+
+	if !lo.Contains(titles, "base") {
+		_, err = db.Exec(ctx, "INSERT INTO harmony_config (title, config) VALUES ('base', '')", "base")
+		if err != nil {
+			return err
+		}
```

Comment:
Maybe this could be done as part of schema creation?

Reply:
I'm torn on the whole "base is required" thing: On one hand, it would be nice to ensure that a layer is always ran that *could* cover the basics, so I made "base" layer always run even without CLI config, and it won't start without one. On the other hand, it's another hurdle that may not mean anything. I could more easily remove the requirement if you don't think it's worthwhile. Let me know your thoughts here.

Reply:
UX: 
Guide help says use 'base'
A naive run is non-destructive by writing a new layer
Either of these populates base with something that will work. 

Ignoring the migrator & doing DiY config leaves you with a start-up error: "no base found".

The result should be a straightforward way to understand that a base layer exists OR a complete ignoring of it.